### PR TITLE
Storybook input fields

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
This PR will add:

- Input component with the following variants: default, error, display, disabled
- Label component that wraps the input field
- Input component stories
- Input snapshot tests

### CODE REVIEW

**Comment by BouweCeunen** [comment]
It seems that the `label` doesn't change on storybook when I edit it in the Control, is this intended? it does work with the placeholder.

**Comment by BartDeCaluwe** [comment]
> It seems that the `label` doesn't change on storybook when I edit it in the Control, is this intended? it does work with the placeholder.

Yeah it has nothing to do with the component but more with how Storybook works.
The "story" params are passed through the Input component and not the Label component. 
But I'll take a look to see if I can fix that.

**Comment by RuvendeGroote** [comment]
image
it's seems like something got broken
https://2468cf26.credix-app-v2.pages.dev/?path=/story/button--primary

**Comment by RuvendeGroote** [comment]
if it's actually broken: shouldn't our snapshot tests pick this up?

**Comment by BartDeCaluwe** [comment]
> if it's actually broken: shouldn't our snapshot tests pick this up?

It depends on what's actually wrong. Snapshot tests check if the "markup" of your component has changed. If for example I break some css class (which is what I'm guessing has happened) then the markup hasn't changed but the look of the component has.

One solution for this would be to use [Chromatic](https://www.chromatic.com/) (made by the same company behind storybook) which basically compares two screenshots of a component and diffs them. 
We could definitely look into this as I believe it's free for OSS.

**Comment by RuvendeGroote** [comment]
> Member

alright, for something visual like a component library I think this is a good idea. is chromatic your preferred choice or just something you looked up right now? I (or you if you want) will add a ticket to set this up

**Comment by BartDeCaluwe** [comment]
> alright, for something visual like a component library I think this is a good idea. is chromatic your preferred choice or just something you looked up right now? I (or you if you want) will add a ticket to set this up

It's not something I've researched, and I've never used something similar before. It's just what Storybook recommends, as they integrate nicely. 
> "Chromatic tests map 1-to-1 with your stories. Whenever you update a story your tests are synced. This means visual tests are painless to keep up to date."

As per their docs. It would surprise me if there is an alternative that would integrate that well, but I can definitely look into it. (I'll make the ticket)

File: __tests__/__mocks__/matchMedia.mock.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Object.defineProperty(window, "matchMedia", {
```
Comment (RuvendeGroote) [inline __tests__/__mocks__/matchMedia.mock.js:1]: why are we doing this?

**Review by RuvendeGroote — DISMISSED** [review]

File: __tests__/__mocks__/matchMedia.mock.js (hunk 1 above)
Comment (BartDeCaluwe) [reply __tests__/__mocks__/matchMedia.mock.js:1]: AntDesign uses `matchMedia` for what I assume is responsiveness reasons (as `matchMedia` returns a `MediaQueryList`). But this is something that is not supported in JSDOM (which is the DOM implementation that Jest uses). This causes an error.
Jest has an official "fix" which is what that code does.
More can be read [here](https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom)

**Review by BouweCeunen — DISMISSED** [review]

**Review by RuvendeGroote — APPROVED** [review]

**Review by BouweCeunen — APPROVED** [review]

**Pull request merged**